Entity/LedgerEntryRepository.php

```diff
@@ -403,102 +403,53 @@ public function getAlternateCampaignSourceData($params, $bySource = false, $cach
-                'ccss.campaign_id',
-                'ccss.contactclient_id',
-                'SUM(IF(ccss.type = "rejected", 1, 0)) AS rejected',
-                'SUM(IF(ccss.type = "converted", 1, 0)) AS converted'
-            )
-            ->from(MAUTIC_TABLE_PREFIX.'contactclient_stats', 'ccss')
-            ->where('ccss.contact_id IN (:leads)')
-            ->groupBy('ccss.contact_id', 'ccss.campaign_id', 'ccss.contactclient_id');
-
-        // Main Query
-        $statBuilder = $this->getEntityManager()->getConnection()->createQueryBuilder();
-        $statBuilder
-            ->select(
-                'COUNT(l.id) AS received',
-                'SUM(cc.rejected) AS rejected',
-                'SUM(cc.converted) AS converted',
-                'SUM(cl.revenue) AS revenue',
-                'cal.campaign_id',
+                'COUNT(cc.contact_id ) AS received',
+                'SUM(IF(cc.type = "rejected", 1, 0)) AS rejected',
+                'SUM(IF(cc.type = "converted", 1, 0)) AS converted',
+                'SUM(cc.attribution) AS revenue',
+                'cc.campaign_id',
                 'lu.utm_source AS utm_source',
                 'c.is_published',
                 'c.name as campaign_name',
                 'cc.contactclient_id as contactclient_id',
                 '0 as cost'
             )
-            ->from(MAUTIC_TABLE_PREFIX.'leads', 'l')
-            ->where('l.date_added BETWEEN :dateFrom AND :dateTo')
-            ->groupBy('cal.campaign_id', 'cc.contactclient_id', 'lu.utm_source')
-            ->setParameter('leads', $leads, Connection::PARAM_STR_ARRAY)
+            ->from(MAUTIC_TABLE_PREFIX.'contactclient_stats', 'cc')
+            ->where('cc.campaign_id IN(:campaignIds)')
+            ->andWhere('cc.date_added BETWEEN :dateFrom AND :dateTo')
+            ->groupBy('cc.campaign_id', 'cc.contactclient_id', 'lu.utm_source')
+            ->setParameter('campaignIds', $campaignIds, Connection::PARAM_STR_ARRAY)
             ->setParameter('dateFrom', $params['dateFrom'])
-            ->setParameter('dateTo', $params['dateTo']);
-        $statBuilder
-            ->leftJoin('l', MAUTIC_TABLE_PREFIX.'campaign_leads', 'cal', 'cal.lead_id = l.id')
-            ->leftJoin('l', MAUTIC_TABLE_PREFIX.'lead_utmtags', 'lu', 'l.id = lu.lead_id')
-            ->innerJoin(
-                'l',
-                '('.$clientstatBuilder->getSQL().')',
-                'cc',
-                'l.id = cc.contact_id AND cc.campaign_id = cal.campaign_id'
-            )
-            ->leftJoin(
-                'l',
-                '('.$ledgerBuilder->getSQL().')',
-                'cl',
-                'l.id = cl.contact_id AND cl.object_id = cc.contactclient_id'
-            )
-            ->leftJoin('l', MAUTIC_TABLE_PREFIX.'campaigns', 'c', 'cl.campaign_id = c.id');
+            ->setParameter('dateTo', $params['dateTo'])
+            ->leftJoin('cc', MAUTIC_TABLE_PREFIX.'lead_utmtags', 'lu', 'cc.contact_id = lu.lead_id')
```

Comment:
Each lead can (and often does) have more than one lead_utmtags entity. 
This query doesn't appear to account for that.